### PR TITLE
Whitelist "data:" in legacy CSP headers

### LINKED ISSUE #19425: CSP for images for legacy apps doesn't match the one in AppFramework
https://github.com/owncloud/core/blob/d1f0ff372e8c05ced760d4e21060e8d8db9c1963/lib/private/response.php#L250

```
'img-src *; '
```

https://github.com/owncloud/core/blob/d1f0ff372e8c05ced760d4e21060e8d8db9c1963/lib/private/response.php#L250

```
private $allowedImageDomains = [
	'\'self\'',
	'data:',
];
```

@LukasReschke

### PR DESCRIPTION
**Pull request by oparoz**
Fixes #19425 

Policy is now `'img-src * data:;`

@LukasReschke

### CODE REVIEW

**Comment by LukasReschke** [comment]
This policy is in there as such since 2014. I'd more consider the apps buggy that do not comply with these…

Ref https://github.com/owncloud/core/commit/3cd32dcb7c368e48a5a7bf510864c8ddd9be85e8

File: lib/private/response.php [hunk 1]
```diff
@@ -247,7 +247,7 @@ public static function addSecurityHeaders() {
 			. 'script-src \'self\' \'unsafe-eval\'; '
 			. 'style-src \'self\' \'unsafe-inline\'; '
 			. 'frame-src *; '
-			. 'img-src *; '
+			. 'img-src \'self\' data:; '
```
Comment (LukasReschke) [inline lib/private/response.php]: Will break stuff. Also add `*` then.

**Comment by oparoz** [comment]
> This policy is in there as such since 2013. I'd more consider the apps buggy that do not comply with these…

Well, the specs says that `*` is not enough

> If the source expression a consists of a single U+002A ASTERISK character (*), and url’s scheme is not one of blob, data, filesystem, then return does match

> As defined above, special URL schemes that refer to specific pieces of unique content, such as "data:", "blob:" and "filesystem:" are excluded from matching a policy of * and must be explicitly listed.

The browsers didn't care, until now.

File: lib/private/response.php (hunk 1 above)
Comment (oparoz) [inline lib/private/response.php]: Yes, I wasn't sure if it needed to be as restrictive as the AppFramework

**Comment by oparoz** [comment]
Let's get some reviewers in: @Xenopathic @rullzer @BernhardPosselt @nickvergessen @DeepDiver1975 @icewind1991 @PVince81 @MorrisJobke

**Comment by LukasReschke** [comment]
> Let's get some reviewers in: @Xenopathic @rullzer @BernhardPosselt @nickvergessen @DeepDiver1975 @icewind1991 @PVince81 @MorrisJobke

I guess that's about 7 too many :speak_no_evil:

**Comment by LukasReschke** [comment]
Shouldn't harm :+1:

**Comment by oparoz** [comment]
> I guess that's about 7 too many :speak_no_evil:

Maybe I should post something on the devel mailing list ;)

**Comment by MorrisJobke** [comment]
:+1: tested and works

**Pull request merged**